solidity/random-beacon/tasks/register-operator.ts

```diff
@@ -0,0 +1,43 @@
+import { task, types } from "hardhat/config"
+
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+
+task("register-operator", "Registers an operator")
+  .addParam("owner", "Stake Owner address", undefined, types.string)
+  .addParam("provider", "Staking Provider", undefined, types.string)
+  .addParam("operator", "Staking Operator", undefined, types.string)
+  .addParam(
+    "application",
+    "Name of Application Contract",
+    undefined,
+    types.string
+  )
```

Comment:
Given that this task is located in `random-beacon` and it is not a generic, cross-project task, there is only one application contract for which the operator can be registered: the `RandomBeacon` contract. I think we should not accept `application` parameter and just assume it's `RandomBeacon`. Same for the sibling task in `ecdsa` and `WalletRegistry`.

Reply:
The same comment applies to `increase-authorization.ts`.

Reply:
The problem is that neither `random-beacon` tasks, nor `ecdsa` tasks know nothing about the contract deployments when we temporarily `git clone` tbtc and deploy all the contracts from the cloned tbtc directory. That cloned tbtc directory will include `random-beacon, ecdsa, and tbtc` contracts and all the artifacts. [This PR](https://github.com/keep-network/tbtc-v2/pull/388) include `mint, stake, increase-authorization, and register-operator` tasks and in order to initialize both applications `beacon` & `ecdsa` running the same tasks, we need to make them accept `application` parameter.

I know it's not super elegant, but we have 2 different repos and 3 different HardHat projects, where `tbtc` depends on `ecdsa` which depends on `beacon`. Some other simple solutions is to have multiple `initialize.sh` files. I'll try to look at other possible solutions to make it more elegant and see what's out there.

Reply:
We chatted with @nkuba about it and the recommendation to avoid it would be not to use the functionality of hardhat-deploy plugin to deploy "all at once" from ecdsa or tbtc-v2. Instead, deploy beacon, ecdsa, and tbtc-v2 separately and yarn link between them during the deployment. This is how testnet contracts are deployed and how mainnet contracts are deployed.

Then, beacon should have tasks such as mint, stake, authorise, and register operator.
Ecdsa should have just authorise and register.

They will be used from testate provisioning scripts.

Reply:
Ok, makes sense. This change required a bunch of refactoring, so I can't provide just one commit. Please look at the changes done in this PR.